src/modules/Game.class.js

```diff
@@ -50,17 +158,97 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.#status;
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    this.#addNumber();
+    this.#addNumber();
+    this.#status = 'playing';
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.#applyInitialState();
+    this.#score = 0;
+    this.#status = 'idle';
+  }
+
+  #applyInitialState() {
+    this.#state = this.#initialState.map((row) => [...row]);
+  }
+  /**
+   * Checks if any move is possible
+   * @returns {boolean}
+   */
+  #isMovePossible() {
+    const gameCopy = new Game(this.getState(), true);
+
+    return (
+      gameCopy.moveLeft() ||
+      gameCopy.moveRight() ||
+      gameCopy.moveUp() ||
+      gameCopy.moveDown()
+    );
+  }
+  /**
+   * Returns if adding number to state was succesfull
+   * @returns {boolean}
+   */
+  #addNumber() {
+    const number = Math.random() > 0.1 ? 2 : 4;
+    const emptyFields = this.#findEmptyFields();
+
+    if (!emptyFields.length) {
+      return false;
+    }
+
+    const randomIndex = Math.floor(Math.random() * emptyFields.length);
+    const { x, y } = emptyFields[randomIndex];
+
+    this.#state[y][x] = number;
+
+    return true;
```

Comment:
The '#addNumber' method is responsible for adding a new number to the game state. It's a good practice to separate the randomness from the core logic for easier testing. Consider passing the random value or position as a parameter.